hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineManagerV2Impl.java

```diff
@@ -420,10 +392,50 @@ public void scrubPipeline(ReplicationType type, ReplicationFactor factor)
           " since it stays at ALLOCATED stage for " +
           Duration.between(currentTime, p.getCreationTimestamp()).toMinutes() +
           " mins.");
-      finalizeAndDestroyPipeline(p, false);
+      closePipeline(p, false);
+      closeContainersForPipeline(p.getId());
```

Comment:
pipeline in allocated state should not have container associate with it. So I think closeContainersForPipeline is only needed for scrubClosedPipeline.

Reply:
Ok good catch